lib/concurrency/Triv_Refinement.thy

```diff
@@ -1,42 +1,40 @@
 (*
+ * Copyright 2024, Proofcraft Pty Ltd
  * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
  *
  * SPDX-License-Identifier: BSD-2-Clause
  *)
 theory Triv_Refinement
-
-imports
-  "Monads.Trace_More_RG"
-  "Monads.Trace_Strengthen_Setup"
-
+  imports
+    "Monads.Trace_More_RG"
+    "Monads.Trace_Strengthen_Setup"
 begin
 
-text \<open>This is a simple (almost trivial) definition of refinement,
-which simply resolves nondeterminism to a smaller set of options.\<close>
-definition
-  triv_refinement :: "('s,'a) tmonad \<Rightarrow> ('s,'a) tmonad \<Rightarrow> bool"
-where
+text \<open>
+  This is a simple (almost trivial) definition of refinement, which simply resolves nondeterminism
+  to a smaller set of options.\<close>
+definition triv_refinement :: "('s,'a) tmonad \<Rightarrow> ('s,'a) tmonad \<Rightarrow> bool" where
   "triv_refinement aprog cprog = (\<forall>s. cprog s \<subseteq> aprog s)"
 
+lemmas triv_refinement_elemD = triv_refinement_def[THEN iffD1, rule_format, THEN subsetD]
+
+lemma triv_refinement_trans:
```

Comment:
Not sure if we did this before, but it'd be nice to have that `triv_refinement` is a partial order, i.e. reflexive, transitive, and and antisymmetric. We could even instantiate it into the order class which would make a bunch of consequences of that available. Not that we are likely to need it much, but it'd be a nice setup.

Reply:
I'm also almost tempted to rename this into `trace_refinement` if that name is still available. Because it's not trivial at all, it is actually the classical version of refinement on traces (it's just not data refinement).